org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/general/completion/FilePathCompletionParticipant.java

```diff
@@ -1,249 +1,258 @@
+				+ valuePathStartOffset; // first character of URI
+		Range filePathRange = XMLPositionUtility.createRange(startPathOffset, endPathOffset, xmlDocument);
+		String originalValuePath = text.substring(startPathOffset, endPathOffset);
+		// ex: valuePath="file:///C:/fold"
+		String valuePath = originalValuePath;
+		String slashInAttribute = getFilePathSlash(valuePath);
+
+		boolean hasFileScheme = valuePath.startsWith(FilesUtils.FILE_SCHEME);
+		if (hasFileScheme) {
+			// remove file:// scheme
+			// ex: valuePath="/C:/fold"
+			valuePath = FilesUtils.removeFileScheme(valuePath);
+			if (valuePath.length() == 0 || valuePath.charAt(0) != '/') {
+				// use of 'file://' and the path was not absolute
+				return;
+			}
+			if (isWindows) {
+				// For Windows OS, remove the last '/' from file:///
+				// ex: valuePath="C:/fold"
+				valuePath = valuePath.substring(1, valuePath.length());
+				if (valuePath.length() == 1) {
+					// only '/', so list Windows Drives
+					Range replaceRange = adjustReplaceRange(xmlDocument, filePathRange, originalValuePath, "/");
+					File[] drives = File.listRoots();
+					for (File drive : drives) {
+						createFilePathCompletionItem(drive, replaceRange, response, "/");
+					}
+					return;
+				}
+			}
+		}
+		// On Linux, Mac OS replace '\\' with '/'
+		if (!isWindows) {
+			if ("\\".equals(slashInAttribute)) { // Backslash used in Unix
+				valuePath = valuePath.replace("\\", "/");
+			}
+		}
+
+		// Get IO path from the given value path
+		Path validAttributePath = getPath(valuePath, xmlDocument.getTextDocument().getUri());
+		if (validAttributePath == null) {
+			return;
+		}
+
+		// Get adjusted range for the completion item (insert at end, or overwrite some
+		// existing text in the path)
+		Range replaceRange = adjustReplaceRange(xmlDocument, filePathRange, originalValuePath, slashInAttribute);
+		createNextValidCompletionPaths(validAttributePath, slashInAttribute, replaceRange, response, null);
+	}
+
+	/**
+	 * Returns the IO Path from the given value path.
+	 * 
+	 * @param valuePath  the value path
+	 * @param xmlFileUri the XML file URI where completion has been triggered.
+	 * @return the IO Path from the given value path.
+	 */
+	private static Path getPath(String valuePath, String xmlFileUri) {
+		// the value path is the filepath URI without file://
+		try {
+			Path validAttributePath = FilesUtils.getPath(valuePath);
+			if (!validAttributePath.isAbsolute()) {
+				// Absolute path, use the XML file URI folder as base dirctory.
+				Path workingDirectoryPath = FilesUtils.getPath(xmlFileUri).getParent();
+				validAttributePath = workingDirectoryPath.resolve(validAttributePath).normalize();
+			}
+			if (!".".equals(valuePath) && !valuePath.endsWith("/") && !valuePath.endsWith("\\")) {
+				// ex : C:/folder|/ -> in this case the path is the folder parent (C:)
+				validAttributePath = validAttributePath.getParent();
+			}
+			return Files.exists(validAttributePath) ? validAttributePath : null;

```

Comment:
`Files.exists(validAttributePath)` returns false if my folder name characters like #, !, ", $, etc.
It returns true for spaces, however.

Reply:
For me if I have `$!#/` as folder I have completion for that  `./$!#/SchemaTest.xsd` but Xerces seems not supporting this kind of folder. Have you the same behvaior?

Reply:
Yes, I get completion items for folder names containing symbols:
![image](https://user-images.githubusercontent.com/20326645/86148209-56c1fa00-bac8-11ea-9f75-41bd5f71b1e6.png)

Reply:
Ok that's mean file path completion is working (scope of this PR). But Xerces doens't support this kind of URI because I suppose that using special character is not in the scope of URI specification.